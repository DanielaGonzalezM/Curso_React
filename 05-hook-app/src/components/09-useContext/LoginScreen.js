import React, { useContext } from "react";
import UserContext from "./UserContext";

const LoginScreen = () => {
  const { user, setUser } = useContext(UserContext);
  return (
    <div>
      <h3>LoginScreen</h3>
      <hr />
      
      <button
        className="btn btn-primary"
        onClick={() => setUser({ ...user, id: 123, name: "Daniela" })}
      >
        Login
      </button>
    </div>
  );
};

export default LoginScreen;
