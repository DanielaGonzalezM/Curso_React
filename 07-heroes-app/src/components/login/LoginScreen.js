import React from "react";
import {useNavigate} from 'react-router-dom';

const LoginScreen = () => {

    const navigate = useNavigate();
  
  const handleLogin = () => {
    navigate('/marvel',{
      replace:true
    });
  };
  return (
    <div className="container mt-5">
      <h1>Login</h1>
      <hr />
      <button className="btn btn-primary" onClick={handleLogin}>
        Login
      </button>
    </div>
  );
};

export default LoginScreen;
