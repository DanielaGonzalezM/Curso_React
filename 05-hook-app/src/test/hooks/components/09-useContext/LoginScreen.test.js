import { mount } from "enzyme";
import LoginScreen from "../../../../components/09-useContext/LoginScreen";
import UserContext from "../../../../components/09-useContext/UserContext";

describe("Pruebas en LoginScreen", () => {
  const setUser = jest.fn();

  const wrapper = mount(
    <UserContext.Provider value={{ setUser }}>
      <LoginScreen />
    </UserContext.Provider>
  );

  test("debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe ejecutar el setUser con el argumento esperado", () => {
    wrapper.find("button").prop("onClick")();
    expect(setUser).toHaveBeenCalledWith({
      id: 123,
      name: "Daniela",
    });
  });
});
