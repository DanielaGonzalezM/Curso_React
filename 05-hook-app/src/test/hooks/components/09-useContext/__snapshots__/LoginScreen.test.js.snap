// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en LoginScreen debe mostrarse correctamente 1`] = `
<div>
  <h3>
    LoginScreen
  </h3>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
